extensions/mssql/src/test/util/dataCache.test.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DataItemCache } from '../../util/dataCache';
+import 'mocha';
+import { should } from 'chai'; should();
+import * as sinon from 'sinon';
+
+describe('DataItemCache', function (): void {
+
+	const testCacheItem = 'Test Cache Item';
+	let fetchFunction: sinon.SinonSpy;
+	let dataItemCache: DataItemCache<String>;
+
+	beforeEach(function (): void {
+		fetchFunction = sinon.spy(() => Promise.resolve(testCacheItem))
```

Comment:
I'd rather we just use TypeMoq for this since that's what we use everywhere else and it's better to be consistent. e.g. for the calledOnce/calledTwice checks below you'd just do something like this https://github.com/microsoft/azuredatastudio/blob/master/extensions/azurecore/src/test/azureResource/tree/treeProvider.test.ts#L84

Reply:
Sure. Not quite as elegant of a statement makes sense to change for consistency sake.
![image](https://user-images.githubusercontent.com/4079568/77780991-08f1ad80-7012-11ea-954c-b1e49d2d4542.png)
Updated.